plugins/module_utils/ndi.py

```diff
@@ -0,0 +1,464 @@
+
+    def format_messages(self, messages):
+        result = {}
+        for message in messages:
+            msg = message.get("message")
+            severity = message.get("severity").lower()
+            result[severity] = msg
+        return result
+
+    def query_messages(self, path):
+        obj = self.nd.query_obj(path, prefix=self.prefix)
+        if obj.get("messages") is not None:
+            result = self.format_messages(obj.get("messages"))
+            return result
+
+    def query_compliance_smart_event(self, ig_name, site_name, compliance_epoch_id):
+        ig_base_path = self.event_insight_group_path.format(ig_name, site_name)
+        size = 100
+        path = "{0}/smartEvents?%24epochId={1}".format(ig_base_path, compliance_epoch_id)
+        event_path = path + "&%24size={0}&%24page={1}&%24sort=-severity&category=COMPLIANCE"
+        smart_event = self.query_entry(event_path, size)
+        return smart_event
+
+    def query_msg_with_data(self, ig_name, site_name, path):
+        ig_base_path = self.event_insight_group_path.format(ig_name, site_name)
+        path = "{0}/{1}/{2}".format(ig_base_path, self.compliance_path, path)
+        result = {}
+        obj = self.nd.query_obj(path, prefix=self.prefix)
+        if obj.get("messages") is not None:
+            message = self.format_messages(obj.get("messages"))
+            if len(message) > 0:
+                result["messages"] = message
+        data = obj.get("value")["data"]
+        if len(data) > 0:
+            result["data"] = data
+        return result
+
+    def query_unhealthy_resources(self, ig_name, site_name, compliance_epoch_id):
+        result = {}
+        ig_base_path = self.event_insight_group_path.format(ig_name, site_name)
+        path = "{0}/{1}/eventUnhealthyResources?%24epochId={2}".format(ig_base_path, self.compliance_path, compliance_epoch_id)
+        objs = self.query_data(path)
+        for obj in objs:
+            result[obj.get("bucket")] = {"count": obj.get("count"), "total": obj.get("total")}
+        return result
+
+    def query_pcvs(self, ig_name):
+        pcvs_path = '{0}/{1}/prechangeAnalysis?$sort=-analysisSubmissionTime'.format(self.config_ig_path, ig_name)
+        obj = self.nd.query_obj(pcvs_path, prefix=self.prefix)
+        return obj['value']['data']
+
+    def query_pcv(self, ig_name, site_name, pcv_name):
+        pcv_results = self.query_pcvs(ig_name)
+        if pcv_name is not None and site_name is not None:
+            site_id = self.get_site_id(ig_name, site_name, prefix=self.prefix)
+            pcv_path = '{0}/{1}/fabric/{2}/prechangeAnalysis'.format(self.config_ig_path, ig_name, site_name)
+            pcv_result = self.get_pre_change_result(pcv_results, pcv_name, site_id, pcv_path, prefix=self.prefix)
+        else:
+            self.nd.fail_json(msg="site name and prechange validation job name are required")
+        return pcv_result
+
+    def is_json(self, myjson):
```

Comment:
static method? should this be part of NDI class?

Reply:
Will try to improve when refactoring codebase

Reply:
ok